v2 client passes nil context; should use either context.TODO() or context.Background()

The v2 client `.Query()` wraps `.QueryContext()` by passing a `nil` context value.

This will cause a panic if any code attempts to actually utilize any of the context's method set (`Done()`, `Deadline()`, `Err()`, etc...).

Below is an example of the type of panic this would cause:

```
$ cat nilctx.go
package main

import (
 "context"
 "time"
)

func x(ctx context.Context) {
 select {
 case <-ctx.Done():
 }
}

func main() {
 go x(nil)

 time.Sleep(5 * time.Second)
}
$ go run ./nilctx.go 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x461722]

goroutine 5 [running]:
main.x(0x0, 0x0)
	/tmp/nilctx/nilctx.go:10 +0x22
created by main.main
	/tmp/nilctx/nilctx.go:15 +0x3d
exit status 2
```

--- a/client/v2/client.go
+++ b/client/v2/client.go
@@ -516,7 +516,7 @@ type Result struct {
 
 // Query sends a command to the server and returns the Response.
 func (c *client) Query(q Query) (*Response, error) {
-	return c.QueryCtx(nil, q)
+	return c.QueryCtx(context.Background(), q)
 }
 
 // QueryCtx sends a command to the server and returns the Response.
@@ -591,7 +591,7 @@ func (c *client) QueryCtx(ctx context.Context, q Query) (*Response, error) {
 
 // QueryAsChunk sends a command to the server and returns the Response.
 func (c *client) QueryAsChunk(q Query) (*ChunkedResponse, error) {
-	req, err := c.createDefaultRequest(nil, q)
+	req, err := c.createDefaultRequest(context.Background(), q)
 	if err != nil {
 		return nil, err
 	}

